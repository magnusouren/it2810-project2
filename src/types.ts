export type User = {
  name: string;
  movieList?: Movie[];
};

export type Movie = {
  id: number;
  title: string;
  year: number;
  rating: number;
  poster: string;
};
