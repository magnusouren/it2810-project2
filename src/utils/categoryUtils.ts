import categoryData from '../../storage/categories.json';
import { Category } from '../types';

export const getCategoryById = (id: number): Category | undefined => {
  const categories: Category[] = categoryData;

  return categories.find((category) => category.id === id);
};

export const getCategoryNameById = (id: number): string | undefined => {
  const category = getCategoryById(id);
  return category?.name;
};

export const getCategoryIds = (): number[] => {
  return categoryData.map((category) => category.id);
};

export const getCategories = (): Category[] => {
  return categoryData;
};

export const getCategoryNames = (): string[] => {
  return categoryData.map((category) => category.name);
};

export const getCategoryIdByName = (name: string): number | undefined => {
  const category = categoryData.find((category) => category.name === name);
  return category?.id;
};
