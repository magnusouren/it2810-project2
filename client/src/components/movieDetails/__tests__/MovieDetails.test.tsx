import { screen } from '@testing-library/react';

import { Movie } from '../../../types';
import { renderWithRouterAndUserContext } from '../../../utils/testUtils';
import { MovieDetails } from '../MovieDetails';

const movie: Movie = {
  adult: false,
  backdrop_path: '/mockBackdropURL.jpg',
  genre_ids: [12, 28], // Adventure, Action
  _id: 111111,
  original_language: 'en',
  original_title: 'Foo Bar',
  overview: 'Simple mock overview',
  popularity: 1000,
  poster_path: '/mockPosterURL.jpg',
  release_date: '1970-01-01',
  title: 'Foo Bar',
  video: false,
  vote_average: 5,
  vote_count: 555,
};

describe('MovieDetails', () => {
  beforeEach(() => {
    renderWithRouterAndUserContext(<MovieDetails movie={movie} />);
  });

  it('should match snapshot', () => {
    const { container } = renderWithRouterAndUserContext(<MovieDetails movie={movie} />);
    expect(container).toMatchSnapshot();
  });

  it('renders the movie title and release year', () => {
    const titleElement = screen.getByText(movie.title + ' (' + movie.release_date.split('-')[0] + ')');

    expect(titleElement).toBeDefined();
  });

  it('renders the movie overview', () => {
    const overviewElement = screen.getByText(movie.overview);
    expect(overviewElement).toBeDefined();
  });

  it('renders the movie poster', () => {
    const posterElement = screen.getByAltText(movie.title + ' backdrop poster');
    expect(posterElement).toBeDefined();
  });

  it('renders the movie genres', () => {
    const genreElement = screen.getByText(movie.release_date + ' - Adventure, Action');
    expect(genreElement).toBeDefined();
  });

  it('renders the movie details', () => {
    const ratingElement = screen.getByText(movie.vote_average + ' / 10');
    const voteCountElement = screen.getByText(movie.vote_count);
    const popularityElement = screen.getByText(movie.popularity);
    expect(ratingElement).toBeDefined();
    expect(voteCountElement).toBeDefined();
    expect(popularityElement).toBeDefined();
  });
});
