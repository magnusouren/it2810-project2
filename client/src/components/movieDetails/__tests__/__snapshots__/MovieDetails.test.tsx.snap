// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MovieDetails > should match snapshot 1`] = `
<div>
  <div
    class="_movieDetails_bdc06c"
  >
    <a
      aria-label="link back to movie page"
      class="_goBack_bdc06c"
      href="/"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="ArrowBackIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
        />
      </svg>
       
      <span>
        Movies
      </span>
    </a>
    <img
      alt="Foo Bar backdrop poster"
      class="_backdropPoster_bdc06c"
      loading="lazy"
      src="https://image.tmdb.org/t/p/original//mockBackdropURL.jpg"
    />
    <div
      class="_container_bdc06c"
    >
      <h1
        tabindex="0"
      >
        Foo Bar
         (
        1970
        )
      </h1>
      <div
        class="_bigContainer_5dd437"
        data-testid="watchlist-button"
      >
        <button
          aria-label="add movie to watchlist"
          aria-pressed="false"
          class="MuiButtonBase-root MuiToggleButton-root MuiToggleButton-sizeSmall MuiToggleButton-primary css-1dlyelz-MuiButtonBase-root-MuiToggleButton-root"
          data-testid="watchlist-toggle-button"
          tabindex="0"
          type="button"
          value="check"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium _add_5dd437 css-i4bv87-MuiSvgIcon-root"
            data-testid="BookmarkAddIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M21 7h-2v2h-2V7h-2V5h2V3h2v2h2v2zm-2 14-7-3-7 3V5c0-1.1.9-2 2-2h7c-.63.84-1 1.87-1 3 0 2.76 2.24 5 5 5 .34 0 .68-.03 1-.1V21z"
            />
          </svg>
          <span
            class="_text_5dd437"
          >
            Add to watchlist
          </span>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
      <p
        class="_date_bdc06c"
        tabindex="0"
      >
        1970-01-01
         - 
        Adventure, Action
      </p>
      <div
        class="_content_bdc06c"
      >
        <div
          class="_textContent_bdc06c"
        >
          <h2
            class="_subtitle_bdc06c"
            tabindex="0"
          >
            About:
          </h2>
          <p
            class="_overview_bdc06c"
            tabindex="0"
          >
            Simple mock overview
          </p>
        </div>
        <h2
          class="_subtitle_bdc06c"
          tabindex="0"
        >
          Statistics:
        </h2>
        <div
          class="_details_bdc06c"
        >
          <div
            class="_detail_bdc06c"
          >
            <div
              aria-label="Average rating by other users: 5 out of 10"
              class="_score_bdc06c"
              tabindex="0"
            >
              5.0
               / 10
              <div>
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                  data-testid="StarIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 17.27 18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                  />
                </svg>
              </div>
            </div>
            <p>
              Avg. Rating
            </p>
          </div>
          <div
            class="_detail_bdc06c"
          >
            <div
              aria-label="Amount of votes: 555"
              class="_score_bdc06c"
              tabindex="0"
            >
              555
              <div>
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                  data-testid="PeopleAltIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M16.67 13.13C18.04 14.06 19 15.32 19 17v3h4v-3c0-2.18-3.57-3.47-6.33-3.87z"
                    fill-rule="evenodd"
                  />
                  <circle
                    cx="9"
                    cy="8"
                    fill-rule="evenodd"
                    r="4"
                  />
                  <path
                    d="M15 12c2.21 0 4-1.79 4-4s-1.79-4-4-4c-.47 0-.91.1-1.33.24C14.5 5.27 15 6.58 15 8s-.5 2.73-1.33 3.76c.42.14.86.24 1.33.24zm-6 1c-2.67 0-8 1.34-8 4v3h16v-3c0-2.66-5.33-4-8-4z"
                    fill-rule="evenodd"
                  />
                </svg>
              </div>
            </div>
            <p>
              Vote Count
            </p>
          </div>
          <div
            class="_detail_bdc06c"
          >
            <div
              aria-label="Popularity: 1000"
              class="_score_bdc06c"
              tabindex="0"
            >
              1000
              <div>
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                  data-testid="LiveTvIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M21 6h-7.59l3.29-3.29L16 2l-4 4-4-4-.71.71L10.59 6H3c-1.1 0-2 .89-2 2v12c0 1.1.9 2 2 2h18c1.1 0 2-.9 2-2V8c0-1.11-.9-2-2-2zm0 14H3V8h18v12zM9 10v8l7-4z"
                  />
                </svg>
              </div>
            </div>
            <p>
              Popularity
            </p>
          </div>
        </div>
      </div>
      <div
        class="_rating_30a570"
      >
        <h2
          tabindex="0"
        >
          Personal rating:
        </h2>
        <p
          tabindex="0"
        >
          You can rate this movie by clicking on the stars below. Your rating will be saved to your account and is a great way to keep track of your favorite movies.
        </p>
        <h3
          tabindex="0"
        >
          Your selected rating: 
          <span>
            No rating selected
          </span>
        </h3>
        <span
          class="MuiRating-root MuiRating-sizeLarge css-jue3ft-MuiRating-root"
        >
          <span
            class="MuiRating-decimal css-34he1w-MuiRating-decimal"
          >
            <label
              class="css-dqr9h-MuiRating-label"
              for=":r1:"
              style="width: 0%; overflow: hidden; position: absolute;"
            >
              <span
                class="MuiRating-icon MuiRating-iconEmpty css-1c99szj-MuiRating-icon"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                  data-testid="StarBorderIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
                  />
                </svg>
              </span>
              <span
                class="MuiRating-visuallyHidden"
              >
                0.5 Stars
              </span>
            </label>
            <input
              class="MuiRating-visuallyHidden"
              id=":r1:"
              name="star-rating"
              type="radio"
              value="0.5"
            />
            <label
              class="css-dqr9h-MuiRating-label"
              for=":r2:"
            >
              <span
                class="MuiRating-icon MuiRating-iconEmpty css-1c99szj-MuiRating-icon"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                  data-testid="StarBorderIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
                  />
                </svg>
              </span>
              <span
                class="MuiRating-visuallyHidden"
              >
                1 Star
              </span>
            </label>
            <input
              class="MuiRating-visuallyHidden"
              id=":r2:"
              name="star-rating"
              type="radio"
              value="1"
            />
          </span>
          <span
            class="MuiRating-decimal css-34he1w-MuiRating-decimal"
          >
            <label
              class="css-dqr9h-MuiRating-label"
              for=":r3:"
              style="width: 0%; overflow: hidden; position: absolute;"
            >
              <span
                class="MuiRating-icon MuiRating-iconEmpty css-1c99szj-MuiRating-icon"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                  data-testid="StarBorderIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
                  />
                </svg>
              </span>
              <span
                class="MuiRating-visuallyHidden"
              >
                1.5 Stars
              </span>
            </label>
            <input
              class="MuiRating-visuallyHidden"
              id=":r3:"
              name="star-rating"
              type="radio"
              value="1.5"
            />
            <label
              class="css-dqr9h-MuiRating-label"
              for=":r4:"
            >
              <span
                class="MuiRating-icon MuiRating-iconEmpty css-1c99szj-MuiRating-icon"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                  data-testid="StarBorderIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
                  />
                </svg>
              </span>
              <span
                class="MuiRating-visuallyHidden"
              >
                2 Stars
              </span>
            </label>
            <input
              class="MuiRating-visuallyHidden"
              id=":r4:"
              name="star-rating"
              type="radio"
              value="2"
            />
          </span>
          <span
            class="MuiRating-decimal css-34he1w-MuiRating-decimal"
          >
            <label
              class="css-dqr9h-MuiRating-label"
              for=":r5:"
              style="width: 0%; overflow: hidden; position: absolute;"
            >
              <span
                class="MuiRating-icon MuiRating-iconEmpty css-1c99szj-MuiRating-icon"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                  data-testid="StarBorderIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
                  />
                </svg>
              </span>
              <span
                class="MuiRating-visuallyHidden"
              >
                2.5 Stars
              </span>
            </label>
            <input
              class="MuiRating-visuallyHidden"
              id=":r5:"
              name="star-rating"
              type="radio"
              value="2.5"
            />
            <label
              class="css-dqr9h-MuiRating-label"
              for=":r6:"
            >
              <span
                class="MuiRating-icon MuiRating-iconEmpty css-1c99szj-MuiRating-icon"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                  data-testid="StarBorderIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
                  />
                </svg>
              </span>
              <span
                class="MuiRating-visuallyHidden"
              >
                3 Stars
              </span>
            </label>
            <input
              class="MuiRating-visuallyHidden"
              id=":r6:"
              name="star-rating"
              type="radio"
              value="3"
            />
          </span>
          <span
            class="MuiRating-decimal css-34he1w-MuiRating-decimal"
          >
            <label
              class="css-dqr9h-MuiRating-label"
              for=":r7:"
              style="width: 0%; overflow: hidden; position: absolute;"
            >
              <span
                class="MuiRating-icon MuiRating-iconEmpty css-1c99szj-MuiRating-icon"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                  data-testid="StarBorderIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
                  />
                </svg>
              </span>
              <span
                class="MuiRating-visuallyHidden"
              >
                3.5 Stars
              </span>
            </label>
            <input
              class="MuiRating-visuallyHidden"
              id=":r7:"
              name="star-rating"
              type="radio"
              value="3.5"
            />
            <label
              class="css-dqr9h-MuiRating-label"
              for=":r8:"
            >
              <span
                class="MuiRating-icon MuiRating-iconEmpty css-1c99szj-MuiRating-icon"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                  data-testid="StarBorderIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
                  />
                </svg>
              </span>
              <span
                class="MuiRating-visuallyHidden"
              >
                4 Stars
              </span>
            </label>
            <input
              class="MuiRating-visuallyHidden"
              id=":r8:"
              name="star-rating"
              type="radio"
              value="4"
            />
          </span>
          <span
            class="MuiRating-decimal css-34he1w-MuiRating-decimal"
          >
            <label
              class="css-dqr9h-MuiRating-label"
              for=":r9:"
              style="width: 0%; overflow: hidden; position: absolute;"
            >
              <span
                class="MuiRating-icon MuiRating-iconEmpty css-1c99szj-MuiRating-icon"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                  data-testid="StarBorderIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
                  />
                </svg>
              </span>
              <span
                class="MuiRating-visuallyHidden"
              >
                4.5 Stars
              </span>
            </label>
            <input
              class="MuiRating-visuallyHidden"
              id=":r9:"
              name="star-rating"
              type="radio"
              value="4.5"
            />
            <label
              class="css-dqr9h-MuiRating-label"
              for=":ra:"
            >
              <span
                class="MuiRating-icon MuiRating-iconEmpty css-1c99szj-MuiRating-icon"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                  data-testid="StarBorderIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
                  />
                </svg>
              </span>
              <span
                class="MuiRating-visuallyHidden"
              >
                5 Stars
              </span>
            </label>
            <input
              class="MuiRating-visuallyHidden"
              id=":ra:"
              name="star-rating"
              type="radio"
              value="5"
            />
          </span>
          <span
            class="MuiRating-decimal css-34he1w-MuiRating-decimal"
          >
            <label
              class="css-dqr9h-MuiRating-label"
              for=":rb:"
              style="width: 0%; overflow: hidden; position: absolute;"
            >
              <span
                class="MuiRating-icon MuiRating-iconEmpty css-1c99szj-MuiRating-icon"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                  data-testid="StarBorderIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
                  />
                </svg>
              </span>
              <span
                class="MuiRating-visuallyHidden"
              >
                5.5 Stars
              </span>
            </label>
            <input
              class="MuiRating-visuallyHidden"
              id=":rb:"
              name="star-rating"
              type="radio"
              value="5.5"
            />
            <label
              class="css-dqr9h-MuiRating-label"
              for=":rc:"
            >
              <span
                class="MuiRating-icon MuiRating-iconEmpty css-1c99szj-MuiRating-icon"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                  data-testid="StarBorderIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
                  />
                </svg>
              </span>
              <span
                class="MuiRating-visuallyHidden"
              >
                6 Stars
              </span>
            </label>
            <input
              class="MuiRating-visuallyHidden"
              id=":rc:"
              name="star-rating"
              type="radio"
              value="6"
            />
          </span>
          <span
            class="MuiRating-decimal css-34he1w-MuiRating-decimal"
          >
            <label
              class="css-dqr9h-MuiRating-label"
              for=":rd:"
              style="width: 0%; overflow: hidden; position: absolute;"
            >
              <span
                class="MuiRating-icon MuiRating-iconEmpty css-1c99szj-MuiRating-icon"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                  data-testid="StarBorderIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
                  />
                </svg>
              </span>
              <span
                class="MuiRating-visuallyHidden"
              >
                6.5 Stars
              </span>
            </label>
            <input
              class="MuiRating-visuallyHidden"
              id=":rd:"
              name="star-rating"
              type="radio"
              value="6.5"
            />
            <label
              class="css-dqr9h-MuiRating-label"
              for=":re:"
            >
              <span
                class="MuiRating-icon MuiRating-iconEmpty css-1c99szj-MuiRating-icon"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                  data-testid="StarBorderIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
                  />
                </svg>
              </span>
              <span
                class="MuiRating-visuallyHidden"
              >
                7 Stars
              </span>
            </label>
            <input
              class="MuiRating-visuallyHidden"
              id=":re:"
              name="star-rating"
              type="radio"
              value="7"
            />
          </span>
          <span
            class="MuiRating-decimal css-34he1w-MuiRating-decimal"
          >
            <label
              class="css-dqr9h-MuiRating-label"
              for=":rf:"
              style="width: 0%; overflow: hidden; position: absolute;"
            >
              <span
                class="MuiRating-icon MuiRating-iconEmpty css-1c99szj-MuiRating-icon"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                  data-testid="StarBorderIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
                  />
                </svg>
              </span>
              <span
                class="MuiRating-visuallyHidden"
              >
                7.5 Stars
              </span>
            </label>
            <input
              class="MuiRating-visuallyHidden"
              id=":rf:"
              name="star-rating"
              type="radio"
              value="7.5"
            />
            <label
              class="css-dqr9h-MuiRating-label"
              for=":rg:"
            >
              <span
                class="MuiRating-icon MuiRating-iconEmpty css-1c99szj-MuiRating-icon"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                  data-testid="StarBorderIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
                  />
                </svg>
              </span>
              <span
                class="MuiRating-visuallyHidden"
              >
                8 Stars
              </span>
            </label>
            <input
              class="MuiRating-visuallyHidden"
              id=":rg:"
              name="star-rating"
              type="radio"
              value="8"
            />
          </span>
          <span
            class="MuiRating-decimal css-34he1w-MuiRating-decimal"
          >
            <label
              class="css-dqr9h-MuiRating-label"
              for=":rh:"
              style="width: 0%; overflow: hidden; position: absolute;"
            >
              <span
                class="MuiRating-icon MuiRating-iconEmpty css-1c99szj-MuiRating-icon"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                  data-testid="StarBorderIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
                  />
                </svg>
              </span>
              <span
                class="MuiRating-visuallyHidden"
              >
                8.5 Stars
              </span>
            </label>
            <input
              class="MuiRating-visuallyHidden"
              id=":rh:"
              name="star-rating"
              type="radio"
              value="8.5"
            />
            <label
              class="css-dqr9h-MuiRating-label"
              for=":ri:"
            >
              <span
                class="MuiRating-icon MuiRating-iconEmpty css-1c99szj-MuiRating-icon"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                  data-testid="StarBorderIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
                  />
                </svg>
              </span>
              <span
                class="MuiRating-visuallyHidden"
              >
                9 Stars
              </span>
            </label>
            <input
              class="MuiRating-visuallyHidden"
              id=":ri:"
              name="star-rating"
              type="radio"
              value="9"
            />
          </span>
          <span
            class="MuiRating-decimal css-34he1w-MuiRating-decimal"
          >
            <label
              class="css-dqr9h-MuiRating-label"
              for=":rj:"
              style="width: 0%; overflow: hidden; position: absolute;"
            >
              <span
                class="MuiRating-icon MuiRating-iconEmpty css-1c99szj-MuiRating-icon"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                  data-testid="StarBorderIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
                  />
                </svg>
              </span>
              <span
                class="MuiRating-visuallyHidden"
              >
                9.5 Stars
              </span>
            </label>
            <input
              class="MuiRating-visuallyHidden"
              id=":rj:"
              name="star-rating"
              type="radio"
              value="9.5"
            />
            <label
              class="css-dqr9h-MuiRating-label"
              for=":rk:"
            >
              <span
                class="MuiRating-icon MuiRating-iconEmpty css-1c99szj-MuiRating-icon"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                  data-testid="StarBorderIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
                  />
                </svg>
              </span>
              <span
                class="MuiRating-visuallyHidden"
              >
                10 Stars
              </span>
            </label>
            <input
              class="MuiRating-visuallyHidden"
              id=":rk:"
              name="star-rating"
              type="radio"
              value="10"
            />
          </span>
          <label
            class="MuiRating-label MuiRating-pristine css-dqr9h-MuiRating-label"
          >
            <input
              class="MuiRating-visuallyHidden"
              id="star-rating-empty"
              name="star-rating"
              type="radio"
              value=""
            />
            <span
              class="MuiRating-visuallyHidden"
            >
              Empty
            </span>
          </label>
        </span>
        <p
          class="_info_30a570"
        >
          Click on a star to select/change rating. The rating will be saved to your account.
        </p>
      </div>
    </div>
  </div>
</div>
`;
